trade-finance-manager-api/src/v1/controllers/deal.party-db.test.js

```diff
@@ -53,3 +53,174 @@ describe('getCompany returns company', () => {
     expect(result.data).toEqual(mockCompany);
   });
 });
+
+jest.mock('../api.js', () => ({
+  getPartyDbInfo: jest.fn(),
+  createParty: jest.fn(),
+}));
+
+describe('when AUTOMATIC_SF_CUSTOMER_CREATION_ENABLED is not true', () => {
+  const originalEnv = process.env;
+
+  beforeAll(() => {
+    process.env = {
+      ...originalEnv,
+      AUTOMATIC_SF_CUSTOMER_CREATION_ENABLED: 'false',
+    };
+  });
+
+  describe('getPartyUrn', () => {
+    it('should return an empty string if no companyRegNo is provided', async () => {
+      const result = await api.getPartyUrn({});
+      expect(result).toBe('');
+    });
+  
+    it('should call getPartyDbInfo and return urn', async () => {
+      const { getPartyDbInfo, createParty } = require('../api.js');
+      getPartyDbInfo.mockResolvedValue([{ partyUrn: 'TEST_URN' }]);
+  
+      const companyData = { companyRegNo: '12345678' };
+      
+      const result = await api.getPartyUrn(companyData);
+  
+      expect(getPartyDbInfo).toHaveBeenCalledWith(companyData);
+  
+      expect(createParty).toHaveBeenCalledTimes(0);
```

Comment:
Can we move this into it's own test please i.e "should not call ..." - same in other tests